rspace/src/main/scala/coop/rchain/rspace/trace/Event.scala

```diff
@@ -57,7 +54,9 @@ object COMM {
       peeks: SortedSet[Int],
       produceCounters: (Seq[Produce]) => Map[Produce, Int]
   ): COMM = {
-    val produceRefs = dataCandidates.map(_.datum.source)
+    val produceRefs =
+      dataCandidates.map(_.datum.source).sortBy(p => (p.channelsHash, p.hash, p.persistent))
```

Comment:
Does this solve the replay issue witnessed in comm events with both persistent and nonpersistent sends/receives?

Reply:
Produces are sorted by all fields from `Produce` so it should work with persistent and nonpersistent sends/receives.
https://github.com/rchain/rchain/blob/79060b3fa15b372e5ed233e845a01a7528228597/rspace/src/main/scala/coop/rchain/rspace/trace/Event.scala#L69-L73